test/boost/s3_test.cc

```diff
@@ -581,3 +581,50 @@ void client_access_missing_object(const client_maker_function& client_maker) {
 SEASTAR_THREAD_TEST_CASE(test_client_access_missing_object_minio) {
     client_access_missing_object(make_minio_client);
 }
+
+SEASTAR_THREAD_TEST_CASE(test_object_reupload) {
+    const sstring name(fmt::format("/{}/testobject-{}", tests::getenv_safe("S3_BUCKET_FOR_TEST"), ::getpid()));
+
+    semaphore mem(16 << 20);
+    auto cln = make_minio_client(mem);
+    auto close_client = deferred_close(*cln);
+
+    for (auto i : {1, 2}) {
+        testlog.info("Put object {}, iteration {}", name, i);
+        temporary_buffer<char> data = sstring("1234567890").release();
```

Comment:
```suggestion
        const sstring test_string("1234567890");
        temporary_buffer<char> data = sstring.get();
```
I didn't check `sstring` interface but `get()` should give you the buffer (it might be called `data()).

Reply:
there is no other methods returning `temporary_buffer<char_type>`